pkg/revisions/tag_watcher.go

```diff
@@ -109,15 +133,34 @@ func (p *tagWatcher) IsMine(obj metav1.ObjectMeta) bool {
 
 func (p *tagWatcher) GetMyTags() sets.String {
 	res := sets.New(p.revision)
-	for _, wh := range p.index.Lookup(p.revision) {
+	mwhTags := sets.New[string]()
+	for _, wh := range p.webhooksIndex.Lookup(p.revision) {
 		res.Insert(wh.GetLabels()[label.IoIstioTag.Name])
+		mwhTags.Insert(wh.GetLabels()[label.IoIstioTag.Name])
+	}
+	for _, svc := range p.servicesIndex.Lookup(p.revision) {
+		tagName := svc.GetLabels()[label.IoIstioTag.Name]
+		res.Insert(tagName)
+
+		if mwhTags.Contains(tagName) {
+			log.Warnf("Tag name %s is already defined by a service. Delete the %s-%s MutatingWebhookConfiguration ", tagName, "istio-revision-tag", tagName)
+			mwhTags.Delete(tagName)
+		}
+	}
+
+	for _, mwhTag := range mwhTags {
+		log.Warnf(
+			"Tag name %s is defined using a MutatingWebhookConfiguration, which is deprecated. Run `istioctl tag set %s --revision '%s' to use a service instead",
```

Comment:
I don't think we've agreed to deprecate this yet.

Reply:
Good catch. What about:

```suggestion
			"Tag name %s is defined using a MutatingWebhookConfiguration, which is going to be deprecated. Run `istioctl tag set %s --revision '%s' to use a service instead",
```

?

Reply:
I don't think we should even say that. We should only say it's deprecated when it is actually deprecated, because plans change and this could get missed if that happens.